daemon/create_windows.go

```diff
@@ -80,3 +85,67 @@ func (daemon *Daemon) createContainerOSSpecificSettings(container *container.Con
 	}
 	return nil
 }
+
+// checkImageIsolationCompatiblity checks whether the provided image is compatible
+// with the host config's isolation settings.
+func (daemon *Daemon) checkImageIsolationCompatiblity(hostConfig *containertypes.HostConfig, img *image.Image) error {
+	hostOSV := osversion.Get()
+	isHyperV := hostConfig.Isolation.IsHyperV()
+	// fallback on default daemon isolation if none was explicitly provided:
+	if hostConfig.Isolation.IsDefault() {
+		isHyperV = daemon.defaultIsolation.IsHyperV()
+	}
+	return checkImageCompatibilityForHostIsolation(hostOSV, img.OSVersion, isHyperV)
+}
+
+// checkImageCompatibilityForHostIsolation checks whether the provided `imageOSVersion`
+// should be able to run on the provided host OS version given Hyper-V isolation.
+// Its contained logic can be distilled into:
+// - if imageOS == hostOS => can run under any isolation mode
+// - if imageOS < hostOS => must be run under Hyper-V
+// - if imageOS > hostOS => can be run under Hyper-V only on an RS5 (1809+) host
+// Please see the below:
```

Comment:
My understanding was:
 - imageOS < hostOS => can be run under Hyper-V only on an RS5 (version 1809+, build 17763, WS2019)
 - if imageOS > hostOS => can be run under as process isolated or in Hyper-V only on WS2022 (build 20348) or Windows 11 (build 22000) hosts

As in, Windows Server 2019 and Windows 10 cannot run Windows Server 2022 containers using Hyper-V [link](https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-2019%2Cwindows-10#windows-server-host-os-compatibility)

Reply:
My bad, my initial understanding on the goal was allowing RS5+ to run any image (older or newer) under Hyper-V.

> As in, Windows Server 2019 and Windows 10 cannot run Windows Server 2022 containers using Hyper-V 

FWIW I was able to successfully build and run a 2022-based image on a Windows 10 host (build 19044, November 2021 patch), but I have not yet tried it on a WS2019 which is exactly RS5 to ensure the cutoff point is ok.

Will return after a bit more testing so we can discuss the final version of the logic.